### DISCUSSION TITLE
HOW CAN WE PROCESS MULTIPLE AUDIOS PARALLELY USING 15GB Nvidia GPU

### DISCUSSION THREAD

**Discussion by sudo-tahirsaeed** [openai/whisper#1996] [Q&A]
Hey is there any way we can process multiple audios parallelly asynchronously with whisper 'large'  i have Nvidia GPU 15GB it takes around 9.7GB after loading model but i have large amount of audios so basically i want to process more than one audio at a time so how many roughly i can process if each audio is 3 min large on avg 
Is there any way to process longer audio at once eliminating 30s limitation ?
And how can we parallely process multiple audios with python here is my usage example code 

audio = whisper.load_audio("/content/audiH.wav")
audio = whisper.pad_or_trim(audio)
mel = whisper.log_mel_spectrogram(audio=audio, n_mels=128).to(model.device)

_, probs = model.detect_language(mel)

print(f"Detected language: {max(probs, key=probs.get)}")

# decode the audio
options = whisper.DecodingOptions(language="ur",without_timestamps=True)
result = whisper.decode(model, mel, options)

output=[]
a=''
sentences = result.text.split(' ')
for sentence in sentences:
    # if all(char in string for char in (chr(i) for i in range(ord('A'), ord('Z')+1))):
    #     a=a+" "+sentence
    # print(a)
    output.append(sentence)
out = ' '.join(output)
output = '\u202B' + out + '\u202C'
print(output)

**Comment by phineas-pta** [comment]
your gpu cannot hold more than 1 instance of whisper so no

**Reply by sudo-tahirsaeed** [reply]
so we need multiple instances for multiple processing we cant do multiple threads running on single instance?

**Comment by toanhuynhnguyen** [comment]
Use multi-threads
